src/main/haskell/kore/src/Data/Result.hs

```diff
@@ -0,0 +1,66 @@
+{-|
+Module      : Data.Result
+Copyright   : (c) Runtime Verification, 2018
+License     : NCSA
+Maintainer  : thomas.tuegel@runtimeverification.com
+-}
+module Data.Result
+    ( Result (..)
+    , definite
+    , fromResult
+    ) where
+
+import           Control.Applicative
+                 ( Alternative (..) )
+import qualified Control.Monad as Monad
+
+{- | @Result a@ encompasses success, failure, and indeterminacy.
+
+    We may view @Failure@ as indicating a failed attempt, whereas @Unknown@
+    indicates that no attempt was made.
+ -}
+data Result a
+    = Success a  -- ^ successful result producing an @a@
+    | Failure  -- ^ unsuccessful result
+    | Unknown  -- ^ indeterminate result
+    deriving (Eq, Ord, Foldable, Functor, Read, Show, Traversable)
+
+-- | Return the result, or the default value if not successful.
+fromResult
+    :: a  -- ^ default value
+    -> Result a  -- ^ result
+    -> a
+fromResult _a =
+    \case
+        Success _a -> _a
+        _ -> _a
```

Comment:
What do you think about
```
fromResult _ (Success _a) = _a
fromResult default _ = default
```
instead? This avoids the `_a` inside the first arm of the lambda case shadowing the first argument of the function.